binding/rust/src/leopard.rs

```diff
@@ -0,0 +1,404 @@
+
+        // SAFETY: most of the unsafe comes from the `load_library_fn` which is
+        // safe, because we don't use the raw symbols after this function
+        // anymore.
+        let (sample_rate, version) = unsafe {
+            let pv_leopard_init =
+                load_library_fn::<PvLeopardInitFn>(&lib, b"pv_leopard_init")?;
+            let pv_sample_rate = load_library_fn::<PvSampleRateFn>(&lib, b"pv_sample_rate")?;
+            let pv_leopard_version =
+                load_library_fn::<PvLeopardVersionFn>(&lib, b"pv_leopard_version")?;
+
+            let status = pv_leopard_init(
+                access_key.as_ptr(),
+                pv_model_path.as_ptr(),
+                addr_of_mut!(cleopard),
+            );
+            check_fn_call_status(status, "pv_leopard_init")?;
+
+            let version = match CStr::from_ptr(pv_leopard_version()).to_str() {
+                Ok(string) => string.to_string(),
+                Err(err) => {
+                    return Err(LeopardError::new(
+                        LeopardErrorStatus::LibraryLoadError,
+                        format!("Failed to get version info from Leopard Library: {}", err),
+                    ))
+                }
+            };
+
+            (pv_sample_rate(), version)
+        };
+
+        Ok(Self {
+            cleopard,
+            sample_rate,
+            version,
+            vtable: LeopardInnerVTable::new(lib)?,
+        })
+    }
+
+    pub fn process(&self, pcm: &[i16]) -> Result<String, LeopardError> {
+        if pcm.len() == 0 {
+            return Err(LeopardError::new(
+                LeopardErrorStatus::FrameLengthError,
+                format!(
+                    "Audio data must not be empty"
+                ),
+            ));
+        }
+
+        let transcript: String;
+        unsafe {
+            let mut transcript_ptr: *mut c_char = std::ptr::null_mut();
+
+            let status = (self.vtable.pv_leopard_process)(self.cleopard, pcm.as_ptr(), pcm.len() as i32, addr_of_mut!(transcript_ptr));
+
+            check_fn_call_status(status, "pv_leopard_process")?;
+
+            transcript = String::from(CStr::from_ptr(transcript_ptr).to_str().map_err(|_| {
+                LeopardError::new(
+                    LeopardErrorStatus::LibraryError(PvStatus::RUNTIME_ERROR),
+                    "Failed to convert transcript string"
+                )
+            })?);
+
+            (self.vtable.pv_free)(transcript_ptr as *mut c_void);
+        };
+
+        Ok(transcript)
+    }
+
+    pub fn process_file<P: AsRef<Path>>(&self, audio_path: P) -> Result<String, LeopardError> {
+        if !audio_path.as_ref().exists() {
+            return Err(LeopardError::new(
+                LeopardErrorStatus::ArgumentError,
+                format!(
+                    "Could not find the audio file at `{}`",
+                    audio_path.as_ref().display()
+                ),
+            ));
+        }
+
+        let transcript: String;
```

Comment:
```suggestion
        let transcript = String::new();
```
Let's not leave any uninitialized variables